### PR TITLE
psgd simple

### PR DESCRIPTION
**Pull request by ranjan1990**

### CODE REVIEW

**Comment by rcurtin** [comment]
This looks a lot better, thanks for taking the time to redo it!  I'll add a couple comments over the next few days.

File: src/mlpack/core/optimizers/parallel_sgd/sgdp_impl.hpp [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+
+
+  //vector of iterate. length of tIterate is same as number of threads available.
+  std::vector<arma::mat> tIterate; 
+
+  //initializing each element of tIterate with initial iterate value 
+  for(size_t i=0;i<num_thread;i++)
+  {
+    tIterate.push_back(iterate);
+  }
```
Comment (rcurtin) [inline src/mlpack/core/optimizers/parallel_sgd/sgdp_impl.hpp]: You could probably do this initialization in parallel too.

File: src/mlpack/core/optimizers/parallel_sgd/sgdp_impl.hpp [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+     overallObjective=0;
+     for (size_t i = 0; i < numFunctions; ++i)
+     {
+       overallObjective += function.Evaluate(sumIterate,i);
+     }
+
+     if (!(std::isnan(overallObjective) || std::isinf(overallObjective)))
+     {
+       sumIterate+=tIterate[i];
```
Comment (rcurtin) [inline src/mlpack/core/optimizers/parallel_sgd/sgdp_impl.hpp]: Do you think we could use an OpenMP reduction operator here?  I think there might be some difficulty; I have not thought it through too in-depth so I don't know if it's completely feasible.

File: src/mlpack/core/optimizers/parallel_sgd/sgdp_impl.hpp (hunk 1 above)
Comment (ranjan1990) [inline src/mlpack/core/optimizers/parallel_sgd/sgdp_impl.hpp]: Yes It can be done.

**Comment by ranjan1990** [comment]
Thank you Ryan for commenting. Little bit busy these days. I will look into this whenever I get free time.

**Comment by rcurtin** [comment]
Hi @ranjan1990, I'm going to close this for inactivity now.  If you'd ever like to handle the issues, we can reopen this and finish it.

**Pull request closed without merging**